Source/AwsCommonRuntimeKit/auth/signing/SigningConfig.swift

```diff
@@ -4,20 +4,16 @@
 import AwsCAuth
 
 // TODO: verify callback logic, fix pointers, and maybe error handling
-public struct SigningConfig {
+public class SigningConfig: CStruct {
```

Comment:
why change these all to `class`?

Reply:
I thought classes have additional features, so why not. Plus, it will help us avoid needless copies by value. I will do a bit of dive deep on Struct vs Classes and make it consistent.

Reply:
Thanks, turning it to struct because they are thread safe and recommeded by Swift.